backend/src/users/user-relationship/user-relationship.gateway.ts

```diff
@@ -0,0 +1,105 @@
+import {
+	MessageBody,
+	OnGatewayConnection,
+	OnGatewayDisconnect,
+	SubscribeMessage,
+	WebSocketGateway,
+	WebSocketServer,
+} from '@nestjs/websockets';
+import { Server, Socket } from 'socket.io';
+import { AuthService } from '../../auth/auth.service';
+import { UserService } from '../user/user.service';
+
+type RelationshipObservers = Record<number, { rooms: string[] }>;
+
+type RelationshipRoom = {
+	source: number;
+	target: number;
+	id: number;
+};
+
+@WebSocketGateway({
+	namespace: '/user/relationship',
+	cors: {
+		origin: '*',
+	},
+})
+export class UserRelationshipGateway
+	implements OnGatewayConnection, OnGatewayDisconnect
+{
+	constructor(
+		private readonly authService: AuthService,
+		private userService: UserService,
+	) {}
+
+	@WebSocketServer()
+	server: Server;
+	public relationshipObservers: RelationshipObservers = {};
+
+	jwtCookieFromHandshakeString(string: string) {
+		return string
+			.split(' ')
+			.find((cookie) => cookie.startsWith('jwt='))
+			?.slice(4);
+	}
```

Comment:
you call this from the socketService

Reply:
This is a statement hehe
I copied it from socketService, I didn't want to instantiate a socketService for only this purpose, lazy I know.
Added it to socket.utils.ts and replaced usage everywhere